freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +120,256 @@ def _get_requests_session(cls, retry_options={}):
                     mmd.get_module_name(),
                     mmd.get_stream_name(),
                 )
                 req_session = self._get_requests_session()
                 response = req_session.get(content_index_url)
-                status_code = response.status_code
-                if status_code == HTTPStatus.OK:
+                if response.ok:
                     images_info = response.json().get("Images", [])
                     for image_info in images_info:
                         image_nvr = image_info["ImageNvr"]
                         is_tagged_auto_build = self._pyxis.image_is_tagged_auto_rebuild(
                             image_nvr
                         )
                         if is_tagged_auto_build:
-                            rebuild_images_list.append(image_nvr)
+                            image_modules_mapping.setdefault(image_nvr, [])
+                            if (
+                                advisory_module_nvr
+                                not in image_modules_mapping[image_nvr]
+                            ):
+                                image_modules_mapping[image_nvr].append(
+                                    advisory_module_nvr
+                                )
                 else:
-                    self.log_error("Fetching module %s data failed.", module_nvr)
+                    self.log_error(
+                        "Fetching module %s data failed.", advisory_module_nvr
+                    )
 
-        return rebuild_images_list
+        return image_modules_mapping
 
-    def _filter_images_with_higher_rpm_nvr(self, event, rebuild_images_list):
+    def _filter_images_with_higher_rpm_nvr(self, images):
         """
-        Filter out images which have higher nvr in rebuild_images_list than
+        Filter out images which have higher NVR in rebuild_images than
         the advisory's.
 
-        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
-            is currently handling.
+        :param images list: A list of image NVRS.
+        :return: a list of ContainerImage instances which can be auto rebuilt.
         :rtype: list
-        :return: List of ContainerImage instances which can be auto rebuilt.
         """
-        errata = Errata()
-        errata_rpm_nvrs = errata.get_cve_affected_rpm_nvrs(event.advisory.errata_id)
+        errata_rpm_nvrs = self.errata.get_cve_affected_rpm_nvrs(
+            self.event.advisory.errata_id
+        )
         lb = LightBlue(
             server_url=conf.lightblue_server_url,
             cert=conf.lightblue_certificate,
             private_key=conf.lightblue_private_key,
             event_id=self.current_db_event_id,
         )
-        images = lb.get_images_by_nvrs(rebuild_images_list, rpm_nvrs=errata_rpm_nvrs)
+        images = lb.get_images_by_nvrs(images, rpm_nvrs=errata_rpm_nvrs)
         return images
+
+    def _parse_compose_source(self, source_value, name_stream_only=True):
```

Comment:
This function hides too many things again. It can be also free function since it does not depend on `self`.

Too simplify you can return/yield the split sources from compose object. So at call sites you would have:

```python
name_stream_set = {f'{n}:{s}' for n,s,v in _compose_sources(compose)}
...
mapping = {f'{n}:{s}':f'{n}:{s}:{v}' for n,s,v in _compose_sources(compose)}
name_stream_set = set(mapping.values())
```

The main things to de-duplicate and move to separate function/method are:
- Getting the sources from compose object
- Splitting into n:s:v

Reply:
Got it. Updated